src/main/java/com/google/sps/data/Instance.java

```diff
@@ -0,0 +1,105 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.Optional;
+import com.google.sps.data.EntityConstants;
+
+/** Class that represents an instance. */
+public class Instance implements InstanceInterface {
+  
+  // The name of this instance.
+  private final String instanceName;
+
+  // The state this instance is in. 
+  // https://cloud.google.com/compute/docs/instances/instance-life-cycle
+  private String state;
+
+  // The session associated with this instance. If sessionId is not present,
+  // then the VM is not being used for a session.
+  private Optional<String> sessionId;
```

Comment:
Add setters for state and sessionId since those are the only two properties that can change after creation. Add tests for them too. 
For the Session class, lets add a setter for ipOfVM, as well. Include that in this PR.

Reply:
The session class is on a different branch so I can't add it to the this PR.